src/core_plugins/kibana/public/dashboard/dashboard_strings.js

```diff
@@ -0,0 +1,48 @@
+import { DashboardViewMode } from './dashboard_view_mode';
+import _ from 'lodash';
+
+export class DashboardStrings {
+
+  /**
+   * @param list {Array.<string>}
+   * @returns {string} The list of strings concatenated with commas so it can be used in a message.
+   * E.g. ['a', 'b', 'c'] returns 'a, b, and c'.
+   */
+  static createStringList(list) {
+    const listClone = _.clone(list);
+    const isPlural = list.length > 1;
+    const lastEntry = isPlural ? `, and ${list[list.length - 1]}` : '';
+    if (isPlural) listClone.splice(-1, 1);
+
+    return `${listClone.join(', ')}${lastEntry}`;
+  }
+
+  /**
+   * @param changedFilters {Array.<string>} An optional list of filter types that have changed.
+   * @returns {string} A warning message to display to the user that they are going to lose changes.
+   */
+  static getUnsavedChangesWarningMessage(changedFilters) {
+    const changedFilterList = this.createStringList(changedFilters);
+
+    return changedFilterList ?
+      `Are you sure you want to cancel and lose changes, including changes made to your ${changedFilterList}?` :
+      `Are you sure you want to cancel and lose changes?`;
+  }
+
+  /**
+   * @param title {string} the current title of the dashboard
+   * @param viewMode {DashboardViewMode} the current mode. If in editing state, prepends 'Editing ' to the title.
+   * @param isDirty {boolean} if the dashboard is in a dirty state. If in dirty state, adds (unsaved) to the
+   * end of the title.
+   * @returns {string} A title to display to the user based on the above parameters.
+   */
+  static getDashboardTitle(title, viewMode, isDirty) {
+    const isEditMode = viewMode === DashboardViewMode.EDIT;
+    const unsavedSuffix = isEditMode && isDirty
+      ? ' (unsaved)'
+      : '';
+
+    const displayTitle = `${title}${unsavedSuffix}`;
+    return isEditMode ? 'Editing ' + displayTitle : displayTitle;
+  }
+}
```

Comment:
Yea, this is a personal preference - I like the pattern of namespacing functions.  Here is my reasoning:

- Avoids naming conflicts.
- Provides additional context.
- Easier to locate the file and location of the function when the exported module name matches the file name.
- Easier to locate all places the function is being used, especially when there are similarly named functions.  E.g. searching for `DashboardStrings.getEditPath` as opposed to `getEditPath` which could be a function used in many different contexts.
- Import only one class and then view all available functions via auto completion, as opposed to navigating to that file to see what else is available.

>  I like this because I can immediately grok the role of the file from this pattern.

Without the namespacing, in order to understand the role of the file from the pattern, you have to read through all the functions and note the function names.  If they are namespaced inside a class, then that tells you the role of the file, without you having to scroll through.  

>  Personally, if I were to come across this code, I would see the class keyword and make the assumption it's supposed to be instantiated.

I understand your point, but from that argument, we should never use static functions because it creates the same confusion (do I need to instantiate or not?).  Maybe this is indeed your stance?

I've already been doing this in other places (for better or worse):
https://github.com/elastic/kibana/blob/master/src/core_plugins/kibana/public/dashboard/filter_utils.js
https://github.com/elastic/kibana/blob/master/src/ui/public/utils/string_utils.js

Another option is to do something like this:
```
export const DashboardStrings = {
 createStringList(list) {..},
 getUnsavedChangesWarningMessage(changedFilters) {..},
 getDashboardTitle() {...}
};
```

Though I find the class syntax more readable.

I'm willing to change my style if that's what the team agrees on, we just haven't had that discussion yet.  Perhaps something for the next "Mend it Monday"?